src/services/CertificateGenerationService.ts

```diff
@@ -563,6 +564,50 @@ class CertificateGenerationService {
 
         return defectString;
     }
+    /**
+     * To fetch trailer registration
+     * @param vin The vin of the trailer
+     * @param make The make of the trailer
+     * @returns A payload containing the TRN of the trailer.
+     */
+     public async getTrailerRegistration(vin: string, make: string) {
+        const config: IInvokeConfig = this.config.getInvokeConfig();
+        const invokeParams: any = {
+            FunctionName: config.functions.trailerRegistration.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "GET",
+                path: `${Configuration.getInstance().getTrailerServiceVersion()}/trailers/${vin}?make=${make}`,
+                pathParameters: {
+                    proxy: `${Configuration.getInstance().getTrailerServiceVersion()}/trailers`
+                },
+                queryStringParameters: {
+                    make
+                }
+            }),
+        };
+        const response = await this.lambdaClient.invoke(invokeParams);
+        try {
+             const payload: any = this.lambdaClient.validateInvocationResponse(response);
+             const trailerRegistration = JSON.parse(payload.body) as ITrailerRegistration;
+             return trailerRegistration.trn;
+         } catch (e) {
+             console.error(`Error on fetching vinOrChassisWithMake ${vin+make}`, e);
+             return undefined;
+         }
+    }
+
+    /**
+     * To check if the testResult is valid for fetching Trn.
+     * @param vehicleType the vehicle type
+     * @param testTypes the test type
+     * @param makeAndModel object containing Make and Model
+     * @returns returns if the condition is satisfied else false
+     */
+    private isValidForTrn(vehicleType: string, testTypes: any, makeAndModel: any): boolean {
+       return (makeAndModel && vehicleType === VEHICLE_TYPES.TRL &&  testTypes.testResult !== TEST_RESULTS.PRS);
```

Comment:
Umer and I had a catch up and we need to clarify whether we need the certificates for PRS